spec/api/api_bookmarks_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'spec_helper'
+require 'api/api_helper'
+
+describe "API BookmarksController" do
+  include ApiHelper
+
+  # Override is_archivist so all users are archivists from this point on
+  class User < ActiveRecord::Base
+    def is_archivist?
+      true
+    end
+  end
+
+  bookmark = { id: "123",
+               url: "http://foo.com",
+               author: "Thing",
+               title: "Title Thing",
+               summary: "<p>blah blah blah</p>",
+               fandom_string: "Testing",
+               rating_string: "General Audiences",
+               category_string: ["M/M"],
+               relationship_string: "Starsky/Hutch",
+               character_string: "Starsky,hutch",
+               notes: "<p>Notes</p>",
+               tag_string: "youpi",
+               collection_names: "",
+               private: "0",
+               rec: "0" }
+
+  describe "Valid API bookmark import" do
+    before do
+      mock_external
+      @user = create(:user)
+    end
+
+    after do
+      WebMock.reset!
+    end
+
+    it "should return 200 OK when all bookmarks are created" do
+      post "/api/v1/bookmarks/import",
+           { archivist: @user.login,
+             bookmarks: [ bookmark ]
+           }.to_json,
+           valid_headers
+      assert_equal 200, response.status
+    end
+
+    it "should return 200 OK when no bookmarks are created" do
+      post "/api/v1/bookmarks/import",
+           { archivist: @user.login,
+             bookmarks: [ bookmark ]
+           }.to_json,
+           valid_headers
+      assert_equal 200, response.status
+    end
+
+    it "should return 200 OK when only some bookmarks are created" do
+      post "/api/v1/bookmarks/import",
+           { archivist: @user.login,
+             bookmarks: [ bookmark, bookmark ]
+           }.to_json,
+           valid_headers
+      assert_equal 200, response.status
+    end
+
+    it "should create bookmarks associated with the archivist" do
+      pseud_id = @user.default_pseud.id
+      post "/api/v1/bookmarks/import",
+           { archivist: @user.login,
+             bookmarks: [ bookmark, bookmark ]
+           }.to_json,
+           valid_headers
+      bookmarks = Bookmark.find_all_by_pseud_id(pseud_id)
+      assert_equal bookmarks.count, 2
```

Comment:
I'm confused by these two last specs. I assumed from the first one that duplicate bookmarks wouldn't be created, and the spec was therefore testing that it would return 200 even if only one bookmark were created. Then the second one seems to expect both to be created? In which case the first spec doesn't seem to be testing "when only some are created". 

My gut feeling is an exact duplicate should ideally not be created? So bookmarks.count should be 1 not 2?